fetch/client/src/main/scala/endpoints4s/fetch/future/Endpoints.scala

```diff
@@ -0,0 +1,31 @@
+package endpoints4s.fetch.future
+
+import endpoints4s.fetch
+
+import scala.concurrent.Future
+import scala.concurrent.Promise
+
+trait Endpoints extends fetch.Endpoints with EndpointsWithCustomErrors
+
+trait EndpointsWithCustomErrors extends fetch.EndpointsWithCustomErrors {
+
+  type Result[A] = Future[A]
```

Comment:
To be honest, this `type Result[A]` is not mandatory to have in Scala.js interpreters. I introduced it in the `xhr-client` interpreter to give more control to the end-users (I think `scala.concurrent.Future` introduces significant overhead in Scala.js, so I wanted to provide an escape hatch), but it’s fine to not introduce the abstract type `Result[A]` in the first place in the fetch interpreter, if you think Fetch users will most likely all choose the same `Result` type, ultimately.

Reply:
I followed this structure to allow both `Future` and `js.Thenable` endpoint results, just how `XHR` did it. I guess there is no harm in leaving it like that, unless I misunderstood your point?

Reply:
True, there is no harm. I just saw an opportunity to simplify things.